src/js/App/analytics/segment-analytics.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { createContext, useContext, useEffect, useRef } from 'react';
+import { AnalyticsBrowser } from '@segment/analytics-next';
+import { isProd } from '../../utils';
+import { useSelector } from 'react-redux';
+import { AnyObject } from '../../types';
+
+type SegmentEnvs = 'dev' | 'prod';
+type SegmentModules = 'openshift';
+
+const KEY_FALLBACK = {
+  prod: 'nm7VsnYsBVJ9MqjaVInft69pAkhCXq9Q',
+  dev: 'Aoak9IFNixtkZJRatfZG9cY1RHxbATW1',
+};
+
+function getAdobeVisitorId() {
+  const visitor = window?.s?.visitor;
+  if (visitor) {
+    return visitor.getMarketingCloudVisitorID();
+  }
+
+  return -1;
+}
+
+const getAPIKey = (env: SegmentEnvs = 'dev', module: SegmentModules) =>
+  ({
+    prod: {
+      openshift: 'z3Ic4EtzJtHrhXfpKgViJmf2QurSxXb9',
+    },
+    dev: {
+      openshift: 'A8iCO9n9Ax9ObvHBgz4hMC9htKB0AdKj',
+    },
+  }[env]?.[module] || KEY_FALLBACK[env]);
+
+const registerUrlObserver = () => {
+  /**
+   * We ignore hash changes
+   * Hashes only have frontend effect
+   */
+  let oldHref = document.location.href.replace(/#.*$/, '');
+
+  const bodyList = document.body;
+  const observer = new MutationObserver((mutations) => {
+    mutations.forEach(() => {
+      const newLocation = document.location.href.replace(/#.*$/, '');
+      if (oldHref !== newLocation) {
+        oldHref = newLocation;
+        window?.sendCustomEvent?.('pageBottom');
+        setTimeout(() => {
+          window.segment?.page(window?._segment?.pageOptions);
+        });
+      }
+    });
+  });
+  observer.observe(bodyList, {
+    childList: true,
+    subtree: true,
+  });
+  return observer.disconnect;
+};
+
+export const SegmentContext = createContext<{ ready: boolean; analytics?: AnalyticsBrowser }>({
+  ready: false,
+  analytics: undefined,
+});
+
+export type SegmentProviderProps = {
+  activeModule: string;
+};
+
+export const SegmentProvider: React.FC<SegmentProviderProps> = ({ activeModule, children }) => {
+  const analytics = useRef<AnalyticsBrowser>();
+  const user = useSelector(({ chrome: { user } }: AnyObject) => user);
+  useEffect(() => {
+    const disconnect = registerUrlObserver();
+    return () => disconnect();
+  }, []);
+
+  useEffect(() => {
+    if (activeModule && user) {
+      const newKey = getAPIKey(isProd() ? 'prod' : 'dev', activeModule as SegmentModules);
+      const identityOptions = { cloud_user_id: user.identity.internal.account_id, adobe_cloud_visitor_id: getAdobeVisitorId() };
+      const groupOptions = {
+        account_number: user.identity.account_number,
+        account_id: user.identity.internal.org_id,
+      };
+      if (!analytics.current) {
+        analytics.current = AnalyticsBrowser.load({ writeKey: newKey }, { initialPageview: false });
+        window.segment = analytics.current;
+        analytics.current.identify(user.identity.internal.account_id, identityOptions);
+        analytics.current.group(user.identity.account_number, groupOptions);
+        analytics.current.page();
```

Comment:
We should strip "/beta" out of the URL's so we're tracking the canonical pages (since most pages are available at both URLs), and just track `beta` as metadata about the page. Plus `module` would probably be pretty useful metadata. Finally, we also need to add `groupId` as context to pretty much all the calls that aren't the group call itself just to be safe and ensure compatibility with destinations that might expect group ID and user ID in every call.

```
function getPagePath() {
  return window.location.pathname.replace(/^\/beta(\/)?/i, '/');
}
function getPagePathSegment(n) {
  const pathSegment = getPagePath().split('/');
  return pathSegment[n] ? pathSegment[n] : '';
}
function getPageURL() {
  const url = window.location.href;
  if (isBeta()) {
    url = url.replace(/\/beta(\/)?/);
  }
  const i = url.indexOf('#');
  return i === -1 ? url : url.slice(0, i);
}
//...
const pageProperties = {
  path: getPagePath(), // strip beta from path
  url: getPageURL(), // strip beta and hash from url
  beta: isBeta(), // add beta to page metadata
  module: activeModule
};
const pageContext = {
  context: {
    groupId: user.identity.internal.org_id
  }
};
//...
analytics.current.page(pageProperties, pageContext);
```